rfc/2025-07-22-conditional-enable.md

```diff
@@ -0,0 +1,225 @@
+# Conditional field for enabling resources and modules
+
+## Current status
+
+Issue: https://github.com/opentofu/opentofu/issues/1306
+
+> [!NOTE]
+> Every time we refer to *resources* in this RFC, we're talking about [managed resources](https://opentofu.org/docs/language/resources/), [data sources](https://opentofu.org/docs/language/data-sources/) and [ephemeral resources](https://github.com/opentofu/opentofu/issues/2834).
+
+Right now, OpenTofu supports conditional enable/disable of resources by using `count` as a workaround:
+
+```
+variable "enabled" {
+  default = true
+}
+
+resource "aws_instance" "example" {
+  count = var.enabled ? 1 : 0
+  # ...
+}
+```
+
+This approach brings a few problems, such as adding indexes to resources that would be single instances, making it harder to manage using these indexes.
+
+This RFC proposes a new way to do this in a cleaner and more semantic manner by using a new field on the `lifecycle` block called `enabled`:
+
+```
+resource "aws_instance" "example" {
+  # ...
+
+  lifecycle {
+    enabled = var.enabled
+  }
+}
+```
+
+This is available not only for resources, but for modules too:
+
+```
+module "modcall" {
+  source = "./mod"
+
+  lifecycle {
+    enabled = var.env == "production"
+  }
+}
+```
+
+## Proposed Solution
+
+1. Support for conditional enabling on:
```

Comment:
I think we still need to explicitly mention in the RFC that the default for this `enabled` field is `true`